Tarea 1, Clase 6

Alfred021 opened the pull request:
Tarea 1, clase 6
La hice pidiendo nada mas empezar la pagina el numero de integrantes.

Alfred021 committed:
Tarea 1, Clase 6

Tarea 1, clase 6

fsodano commented on tareas/clase-6/tarea 1, clase 6.html:
```diff
@@ -0,0 +1,36 @@
+                
+        </div>
+
+        <button id='calcular' type ="button">Calcular</button>
```
se consistente con las comillas, usá `""` para html

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.html:
```diff
@@ -0,0 +1,36 @@
+<body>
+    <h1> Grupo Familiar Tarea 1</h1>
+
+    <form id="para-calcular-edades">
```
este nombre no me gusta. capaz sólo `calcular-edades` está mejor

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.html:
```diff
@@ -0,0 +1,36 @@
+    
+    <script type="text/javascript" src="tarea 1, clase 6.js"></script>
+</body>
+</html>
```
acordate de dejar una línea vacía al final de cada archivo, github te avisa mostrándote el circulito rojo

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+
+    for (i = 0; i< miembrosGrupoFamiliar; i++) {
+        CrearMiembros(i);
+    }
```
esto está mal identado, debería estar a la altura de `let miembrosGrupoFamiliar` porque no está dentro de nada

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+    return false
+};
+ 
+function CrearMiembros(numero = 1) {
```
las funciones y variables no van con `PascalCase`, van con `camelCase`, es decir, `crearMiembros`.

En vez de número, yo le pondría `indice`!

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+    }
+
+document.querySelector('#calcular').onclick = function() {
+    let edades = SacarEdades ();
```
`sacar` es muy coloquial, lo que vos querés hacer es `obtenerEdadesMiembros`

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+
+document.querySelector('#calcular').onclick = function() {
+    let edades = SacarEdades ();
+    document.querySelector('#mayor-edad').value = sacarmayorEdad ();
```
`calcularMayorEdad`

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+document.querySelector('#calcular').onclick = function() {
+    let edades = SacarEdades ();
+    document.querySelector('#mayor-edad').value = sacarmayorEdad ();
+    document.querySelector('#menor-edad').value = sacarmenorEdad ();
```
`calcularMenorEdad`

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+    let edades = SacarEdades ();
+    document.querySelector('#mayor-edad').value = sacarmayorEdad ();
+    document.querySelector('#menor-edad').value = sacarmenorEdad ();
+    document.querySelector('#promedio-edad').value = sacarpromedioEdad(edades);
```
`calcularPromedioEdades`, lo que hiciste bien acá es pasar las edades como parámetros!  En los otros 2, estás obteniendo las edades de los miembros otra vez, es mejor que digas que las funciones `calcularMayorEdad` o `calcularMenorEdad` toman como parámetro un array de edades.

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+    borrarIntegrantes ();
+    borrarnumMayor ();
+    borrarnumMenor ();
+    borrarPromedio ();
```
los paréntesis van justo después del nombre de la función: `borrarIntegrantes()`.

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+
+    if (numero >= 1) {
+        acumulador = 0;
+        acumulador + 1;
```
Acá no estás guardando el resultado de esta operación en ningún lado

Alfred021 replied:
listo, lo cambie usando numero

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+    div.className = 'integrante';
+
+    const label = document.createElement('label');
+    label.textContent = 'Edad del integrante Nro: ' + (acumulador);
```
Este acumulador no te da siempre 1? Creo que simplemente podrías usar la variable `numero`, ya que estás pasando el nro del integrante como parámetro.

Imagino que lo que querés evitar es que diga `integrante nro 0`, en cuyo caso podés poner simplemente `'Edad del integrante nro: ' + (numero + 1)`

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+}
+
+function sacarmayorEdad () {
+    let mayorEdad = SacarEdades ();
```
`mayorEdad` va a contener un listado de edades... por lo tanto el nombre no está bien. El nombre sería `edades`o `edadesMiembros` o `listadoEdades`, pero no `mayorEdad`.

Por otro lado, sería mejor que tomes el listado de edades como parámetro.

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+
+function sacarmayorEdad () {
+    let mayorEdad = SacarEdades ();
+    let resultadomayorEdad = Math.max(...mayorEdad);
```
Esto está bien, pero te recomiendo que lo resuelvas con un `for` para pensarlo vos sólo.

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+}
+
+function sacarpromedioEdad(edades) {
+    let contador = 0;
```
Esto sería un `acumulador`

Alfred021 replied:
listo

fsodano commented on tareas/clase-6/tarea 1, clase 6.js:
```diff
@@ -0,0 +1,93 @@
+    return contador / edades.length;
+}
+
+function borrarnumMayor () {
```
bien, sólo fijate que las funciones sean `camelCase`. `borrarNumeroMayor`

Alfred021 replied:
listo

Alfred021 committed:
Cambios sugeridos por fabricio

Se corrigieron los nombres, se reescribieron algunas funciones y otros consejos en general de la revision de fabricio.
